### PR TITLE
Clean up lots of todos

### LINKED ISSUE #829: Pre-release: clean up todos in developer guide
Before final release, go through the docs and clean up any visible TODO items, either by writing the documentation or commenting out the section.

### PR DESCRIPTION
**Pull request by stevenschlansker**
Some deletes, some folding together, some commented out.

Simplify serialization test case

Fixes #829

### CODE REVIEW

**Comment by stevenschlansker** [comment]
This is going to be need to be rebased after #715 lands, I'll take care of that.

File: docs/src/adoc/index.adoc [hunk 1]
```diff
@@ -4040,12 +4032,14 @@ TODO:
...
   annotations.
 ** SqlMethodAnnotatedHandlerDecorator
+////
 
 === Extensions
 
 Jdbi supports generic extensions. SQL Objects are just one implementation of
 that spec. If SQL Objects aren't doing it for you, you can create your own!
 
+////
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: I would comment out the entire section--the intro above is sort of pointless by itself.

File: docs/src/adoc/index.adoc [hunk 2]
```diff
@@ -4069,6 +4064,7 @@ Be careful using this feature! This can become a vector for SQL injection
 attacks if you fail to
 link:https://xkcd.com/327/[sanitize your database inputs^].
 
+////
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: Ditto for this section--the header by itself only teases but doesn't tell the reader anything about it.

File: docs/src/adoc/index.adoc [hunk 3]
```diff
@@ -4214,9 +4211,9 @@ Most users will not need to implement the *ResultProducer* interface.
...
 === Best Practices
 
-TODO:
-
 * Test your SQL Objects (DAOs) against real databases when possible.
+  Jdbi tries to be defensive and fail eagerly when you hold it wrong.
+
 * Use the `-parameters` compiler flag to avoid all those
   `@Bind("foo") String foo` redundant qualifiers in SQL Object method
   parameters.
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: Let's add a `See: <<Automatic parameter naming>>` to this bullet.

File: docs/src/adoc/index.adoc [hunk 4]
```diff
@@ -4233,7 +4230,8 @@ TODO:
 
 === Related Projects
 
-TODO:
+link:https://github.com/opentable/otj-pg-embedded[Embedded Postgres^]
+makes testing against a real database quick and easy.
 
 * arteam/dropwizard-jdbi3
 * arteam/metrics-jdbi3
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: Can we convert these to links with summaries, like otj-pg-embedded?

File: docs/src/test/java/jdbi/doc/TransactionTest.java [hunk 5]
```diff
@@ -158,74 +159,46 @@ default void innerMethodWithADifferentLevel() {}
...
+        insert.accept(handle, 20);
 
+        // Run the following twice in parallel, and synchronize
         ExecutorService executor = Executors.newCachedThreadPool();
         CountDownLatch latch = new CountDownLatch(2);
 
-        // Both of these would calculate 10 + 20 = 30, but that violates serialization!
-        SumAndInsert txn1 = new SumAndInsert(latch, db);
-        SumAndInsert txn2 = new SumAndInsert(latch, db);
+        Supplier<Callable<Integer>> sumAndInsert = () -> () ->
```
Comment (qualidafial) [inline docs/src/test/java/jdbi/doc/TransactionTest.java]: yo dawg, I heard you like lambdas

**Pull request merged**